wgengine/magicsock/magicsock.go

```diff
@@ -2533,6 +2533,32 @@ func (c *Conn) SetDERPMap(dm *tailcfg.DERPMap) {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
+	if debugUseDERPAddr() != "" {
+		// We only need to set the DERPMap once
```

Comment:
There's already an early exit below, this one may not be necessary. It's possible for envknob values to be changed at runtime (via `envknob.Setenv`), so `debugUseDERPAddr()` may return a different value across calls.